lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -205,30 +205,29 @@ SPIRVToLLVMDbgTran::transCompilationUnit(const SPIRVExtInst *DebugInst,
 
   BuilderMap[DebugInst->getId()] = std::make_unique<DIBuilder>(*M);
 
-  if (DebugInst->getExtSetKind() == SPIRVEIS_NonSemantic_Shader_DebugInfo_100) {
-    return BuilderMap[DebugInst->getId()]->createCompileUnit(
-        SourceLang, getFile(Ops[SourceIdx]), CompilerVersion, false, Flags, 0);
-  }
-  if (DebugInst->getExtSetKind() == SPIRVEIS_NonSemantic_Shader_DebugInfo_200) {
-    StringRef Producer = getString(Ops[ProducerIdx]);
-    return BuilderMap[DebugInst->getId()]->createCompileUnit(
-        SourceLang, getFile(Ops[SourceIdx]), Producer, false, Flags, 0);
-  }
-  // TODO: Remove this workaround once we switch to NonSemantic.Shader.* debug
-  // info by default
-  auto Producer = findModuleProducer();
   assert(BuilderMap.size() != 0 && "No debug compile units");
-  if (BuilderMap.size()==1)
+  if (isNonSemanticDebugInfo(DebugInst->getExtSetKind())) {
```

Comment:
Is it possible for some compilation units to use non-semantic debug info and others to not use it?

Reply:
Yes, just like for OpLine / DebugLine feature we can get such module after spirv-link

Reply:
BTW, do we even need changes in the reverse translator?

Reply:
We need these changes because the previous BuildIdentifier/StoragePath generation was put in the wrong path.  I moved it to the NonSemanticDebugInfo path.  I also unified the DebugInfo_100 and DebugInfo_200 paths.

Currently, the code only invokes setBuildIdentifierAndStoragePath() once by checking if BuilderMap.size()==1.  But could this fail if the first compilation unit does not have NonSemanticDebugInfo but subsequent compilation units do have NonSemanticDebugInfo?